src/version-parser.spec.ts

```diff
@@ -0,0 +1,13 @@
+import { getVersionToUse } from './version-parser'
+
+test('should return agent version', async () => {
+    expect(await getVersionToUse("https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-agent/maven-metadata.xml", false)).not.toBe("");
+});
+
+test('should return gradle instrumentator version', async () => {
+    expect(await getVersionToUse("https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-instrumentation-for-gradle/maven-metadata.xml", false)).not.toBe("");
+});
+
+test('should return gradle plugin version', async () => {
+    expect(await getVersionToUse("https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-gradle-plugin/maven-metadata.xml", false)).not.toBe("");
+});
```

Comment:
ideally you would want to run this in CI, right?

Reply:
Is there something special to add if you want to run in a CI?

Reply:
no, nothing, a simple GHA with node setup should work. Something like https://github.com/undefinedlabs/demo-nodejs/blob/master/.github/workflows/test.yml